diff -Nur src_orig/CMakeLists.txt src/CMakeLists.txt
--- src_orig/CMakeLists.txt	2010-05-11 13:02:26.326292641 +0200
+++ src_orig/CMakeLists.txt	2010-05-11 13:02:16.478845124 +0200
@@ -80,6 +80,18 @@
 #
 # When changing any of this remember to update the notes in doc/blender-cmake.txt
 
+  find_package(Freetype)
+  message("Freetype-->${Freetype}<-->${Freetype_INCLUDE_DIR}<-->${Freetype_LIBRARIES}<--")
+  SET(FREETYPE_LIB ${FREETYPE_LIBRARIES})
+  SET(FREETYPE_INC ${FREETYPE_INCLUDE_DIRS})
+
+  find_package(PNG)
+  message("PNG-->${PNG}<-->${PNG_INCLUDE_DIR}<-->${PNG_LIBRARIES}<--")
+  SET(PNG_LIB ${PNG_LIBRARIES})
+  SET(PNG_INC ${PNG_INCLUDE_DIR})
+
+
+
 #-----------------------------------------------------------------------------
 # Load some macros.
 INCLUDE(CMake/macros.cmake)
@@ -88,6 +100,7 @@
 #Platform specifics
 
 IF(UNIX)
+
   IF(WITH_OPENAL)
     INCLUDE(${CMAKE_ROOT}/Modules/FindOpenAL.cmake)
     IF(OPENAL_FOUND)
@@ -123,18 +136,19 @@
     SET(GETTEXT_LIB ${INTL_LIBRARY} ${ICONV_LIBRARY})
   ENDIF(INTL_LIBRARY AND ICONV_LIBRARY)
   
-  FIND_PATH(FREETYPE_INC
-    freetype
-    PATHS
-    /usr/local/include/freetype2
-    /usr/include/freetype2
-    /sw/include/freetype2
-    /opt/local/include/freetype2
-    /opt/csw/include/freetype2
-    /opt/include/freetype2
-    NO_DEFAULT_PATH
-  )
-  SET(FREETYPE_LIB freetype)
+#  FIND_PATH(FREETYPE_INC
+#    freetype
+#    PATHS
+#    /usr/local/include/freetype2
+#    /usr/include/freetype2
+#    /sw/include/freetype2
+#    /opt/local/include/freetype2
+#    /opt/csw/include/freetype2
+#    /opt/include/freetype2
+##    NO_DEFAULT_PATH
+#  )
+#  SET(FREETYPE_LIB freetype)
+
 
   INCLUDE(${CMAKE_ROOT}/Modules/FindPythonLibs.cmake)
   SET(PYTHON_INC "${PYTHON_INCLUDE_PATH}" CACHE STRING "")
@@ -164,11 +178,22 @@
   SET(FFMPEG_LIB avformat avcodec avutil avdevice swscale)
   SET(FFMPEG_LIBPATH ${FFMPEG}/lib)
 
-  SET(JPEG_LIB jpeg)
+  find_package(JPEG)
+  SET(JPEG_LIB ${JPEG_LIBRARIES})
+  SET(JPEG_INC ${JPEG_INCLUDE_DIR})
 
-  SET(PNG_LIB png)
 
-  SET(ZLIB_LIB z)
+  #SET(JPEG_LIB jpeg)
+
+  #SET(PNG_LIB png)
+
+  #SET(ZLIB_LIB z)
+
+  find_package(TIFF)
+
+  message("tiff-->${TIFF_FOUND}<-->${TIFF_INCLUDE_DIR}<-->${TIFF_LIBRARIES}<--")
+  SET(TIFF ${TIFF_LIBRARIES})
+  SET(TIFF_INC ${TIFF_INCLUDE_DIR})
 
   SET(LLIBS "-lXi -lutil -lc -lm -lpthread -lstdc++")
 
