diff -Nur src_orig/CMakeLists.txt src/CMakeLists.txt
--- src_orig/CMakeLists.txt	Fri Feb 12 11:45:00 2010
+++ src/CMakeLists.txt	Thu Feb 25 00:53:40 2010
@@ -60,7 +60,7 @@
 
 # We have some custom .cmake scripts not in the official distribution.
 # Maybe this can be used override existing behavior if needed?
-SET(CMAKE_MODULE_PATH "${OpenSceneGraph_SOURCE_DIR}/CMakeModules;${CMAKE_MODULE_PATH}")
+SET(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${OpenSceneGraph_SOURCE_DIR}/CMakeModules")
 
 
 # Okay, here's the problem: On some platforms, linking against OpenThreads
