diff -Nur src.orig/subversion/mod_dav_svn/util.c src/subversion/mod_dav_svn/util.c
--- src.orig/subversion/mod_dav_svn/util.c	2012-03-30 15:49:44.104444000 +0200
+++ src/subversion/mod_dav_svn/util.c	2012-03-30 15:55:38.102790000 +0200
@@ -627,8 +627,8 @@
         if (errscan->desc == NULL)
             continue;
 
-        if (errscan->save_errno != 0) {
-            errno = errscan->save_errno;
+        if (errscan->aprerr != 0) {
+            errno = errscan->aprerr;
             ap_log_rerror(APLOG_MARK, level, errno, r, "%s  [%d, #%d]",
                           errscan->desc, errscan->status, errscan->error_id);
         }
