diff -Nur D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\download\curl-7.19.7/CMakeLists.txt D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\src/CMakeLists.txt
--- D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\download\curl-7.19.7/CMakeLists.txt	Wed Jul 22 20:09:53 2009
+++ D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\src/CMakeLists.txt	Sun Feb 21 22:20:15 2010
@@ -18,7 +18,7 @@
 # (From Daniel Stenberg) The cmake build selected to run gcc with -fPIC on my box while the plain configure script did not.
 # (From Daniel Stenberg) The gcc command line use neither -g nor any -O options. As a developer, I also treasure our configure scripts's --enable-debug option that sets a long range of "picky" compiler options.
 cmake_minimum_required(VERSION 2.6.2 FATAL_ERROR)
-set(CMAKE_MODULE_PATH "${CMAKE_CURRENT_SOURCE_DIR}/CMake;${CMAKE_MODULE_PATH}")
+set(CMAKE_MODULE_PATH "${CMAKE_MODULE_PATH};${CMAKE_CURRENT_SOURCE_DIR}/CMake")
 include(Utilities)
 
 project( CURL C )
diff -Nur D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\download\curl-7.19.7/lib/CMakeLists.txt D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\src/lib/CMakeLists.txt
--- D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\download\curl-7.19.7/lib/CMakeLists.txt	Wed Jul 22 20:09:53 2009
+++ D:\portable_dev\devel\auto_build\src\libs\Sources\curl\7.19.7\src/lib/CMakeLists.txt	Sun Feb 21 20:40:56 2010
@@ -122,3 +122,36 @@
     set_target_properties(${LIB_NAME} PROPERTIES IMPORT_SUFFIX "_imp.lib")
   endif()
 endif()
+
+#INSTAL first cut
+
+set(INSTALL_INCDIR include)
+set(INSTALL_BINDIR bin)
+
+#this is a placelholder to use under linux to diff 32/64 os versions
+set(LIB_POSTFIX "" )
+
+if(WIN32)
+    set(INSTALL_LIBDIR bin)
+    set(INSTALL_ARCHIVEDIR lib)
+else()
+    set(INSTALL_LIBDIR lib${LIB_POSTFIX})
+    set(INSTALL_ARCHIVEDIR lib${LIB_POSTFIX})
+endif()
+
+
+install(
+    TARGETS ${LIB_NAME}
+    RUNTIME DESTINATION bin COMPONENT libcurl
+    LIBRARY DESTINATION lib COMPONENT libcurl
+    ARCHIVE DESTINATION ${INSTALL_ARCHIVEDIR} COMPONENT libcurl-dev
+)
+
+file(GLOB HEADERS_TO_INSTALL "${CMAKE_SOURCE_DIR}/include/curl/*.h")
+list(APPEND HEADERS_TO_INSTALL ${CURL_BINARY_DIR}/include/curl/curlbuild.h)
+message("installing --> ${HEADERS_TO_INSTALL}" )
+INSTALL(
+    FILES        ${HEADERS_TO_INSTALL}
+    DESTINATION ${INSTALL_INCDIR}/curl
+    COMPONENT libcurl-dev
+)
\ No newline at end of file
