diff -Nur src/CMakeLists.txt src/CMakeLists.txt
--- src/CMakeLists.txt	Sun Feb 21 22:20:15 2010
+++ src/CMakeLists.txt	Mon Feb 22 00:29:25 2010
@@ -816,7 +816,7 @@
   if(CMAKE_USE_OPENSSL AND OPENSSL_FOUND)
     include_directories(${OPENSSL_INCLUDE_DIR})
   endif()
-  if(CMAKE_USE_OPENSSL AND CURL_CONFIG_HAS_BEEN_RUN_BEFORE)
+  if(CMAKE_USE_OPENSSL AND OPENSSL_FOUND)
     target_link_libraries(${TARGET_NAME} ${OPENSSL_LIBRARIES})
     #ADD_DEFINITIONS( -DUSE_SSLEAY )
   endif()
