diff -Nur src.orig/CMakeLists.txt src/CMakeLists.txt
--- src.orig/CMakeLists.txt	Wed Jul 22 20:09:53 2009
+++ src/CMakeLists.txt	Thu Feb 25 09:54:04 2010
@@ -783,12 +783,21 @@
 
 if(HAVE_SOCKLEN_T)
   set(CURL_TYPEOF_CURL_SOCKLEN_T "socklen_t")
+  if(UNIX)
+	SET(CMAKE_EXTRA_INCLUDE_FILES sys/socket.h)
+  else()
+	if(MSVC)
+		SET(CMAKE_EXTRA_INCLUDE_FILES "winsock2.h;ws2tcpip.h")
+	endif()
+  endif()
   check_type_size("socklen_t" CURL_SIZEOF_CURL_SOCKLEN_T)
+  SET(CMAKE_EXTRA_INCLUDE_FILES)
 else()
   set(CURL_TYPEOF_CURL_SOCKLEN_T int)
   set(CURL_SIZEOF_CURL_SOCKLEN_T ${SIZEOF_INT})
 endif()
 
+#message("UUUUUUUUUUUUUUUUU HAVE_SOCKLEN_T -->${HAVE_SOCKLEN_T}<--CURL_SIZEOF_CURL_SOCKLEN_T -->${CURL_SIZEOF_CURL_SOCKLEN_T}<--SIZEOF_INT-->${SIZEOF_INT}<--")
 include(CMake/OtherTests.cmake)
 
 add_definitions(-DHAVE_CONFIG_H)
