diff -Nur src_orig/lib/CMakeLists.txt src/lib/CMakeLists.txt
--- src_orig/lib/CMakeLists_orig.txt	Thu Apr 28 13:08:09 2011
+++ src/lib/CMakeLists.txt	Tue Aug 30 13:59:49 2011
@@ -120,6 +120,10 @@
   if(NOT CURL_STATICLIB)
     # Add "_imp" as a suffix before the extension to avoid conflicting with the statically linked "libcurl.lib"
     set_target_properties(${LIB_NAME} PROPERTIES IMPORT_SUFFIX "_imp.lib")
+  else(NOT CURL_STATICLIB)
+    # Add "_imp" as a suffix before the extension to avoid conflicting with the statically linked "libcurl.lib"
+    set_target_properties(${LIB_NAME} PROPERTIES SUFFIX "_static.lib")
+    set_target_properties(${LIB_NAME} PROPERTIES OUTPUT_NAME "curllib")
   endif()
 endif()
 
